### ISSUE TITLE
Improve sorting in method call expression context

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#36467]
**Description:**
![image](https://user-images.githubusercontent.com/35211477/172437951-bd2614d2-9359-478d-843c-a25aec2807ec.png)

The `payload` completion item should be given the highest precedence. 

```ballerina
service /conversionRates on new http:Listener(9090) {

    Subscriber[] subscribers;

    function init() {
        task:JobId result = check task:scheduleJobRecurByFrequency(new Job(), 60, maxCount = 30);
        self.subscribers = [];
    }

    resource function post register(@http:Payload Subscriber payload) {
        self.subscribers.push(<cursor>)
    }
}

```

**Comment by malinthar** [comment]
The issue is no longer reproducible in the 2201.1.x branch. However the sorting order can be improved to de-prioritize the `Type` completion items in this case.
![image](https://user-images.githubusercontent.com/35211477/176606125-27191c45-f766-4b9d-accb-218ca63bf22f.png)

**Comment by malinthar** [comment]
> The issue is no longer reproducible in the 2201.1.x branch. However the sorting order can be improved to de-prioritize the `Type` completion items in this case. ![image](https://user-images.githubusercontent.com/35211477/176606125-27191c45-f766-4b9d-accb-218ca63bf22f.png)

Related issue https://github.com/ballerina-platform/ballerina-lang/issues/36349

**Comment by malinthar** [comment]
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/36897

**Comment by LakshanWeerasinghe** [comment]
Fixed with #37052
